### PR TITLE
enable syncing while idle on Android M devices

### PR DESCRIPTION
**Pull request by TETRA2000**
#970 #857
Change to use new AlarmManager API when it's available and whitelisted by user.
I think it's better to try fetching email as possible as frequent when user wanted.

I confirmed emails are fetched in this [test method](http://developer.android.com/training/monitoring-device-state/doze-standby.html#testing_doze_and_app_standby) and my daily use with N7(2013).

Sorry if my english is hard to understand.

### CODE REVIEW

**Comment by cketti** [comment]
Please make updating to API 23 a separate commit.

File: k9mail/src/main/java/com/fsck/k9/helper/K9AlarmManager.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import android.annotation.TargetApi;
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.os.Build;
+import android.os.PowerManager;
+
+public class K9AlarmManager {
+    private AlarmManager mAlarmManager;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/helper/K9AlarmManager.java]: Please check the [code style](https://github.com/k9mail/k-9/wiki/CodeStyle) document. New code shouldn't use the "m"  prefix.

**Comment by TETRA2000** [comment]
@cketti Thank you for your review.

I updated my PR.
Please check again.

**Comment by cketti** [comment]
Awesome. Thanks!

**Comment by TETRA2000** [comment]
@cketti Thank you for merging!!

**Pull request merged**